src/pocketmine/network/mcpe/NetworkBinaryStream.php

```diff
@@ -74,6 +77,62 @@ public function putUUID(UUID $uuid) : void{
 		$this->putLInt($uuid->getPart(2));
 	}
 
+	public function getSkin() : Skin{
+		$skinId = $this->getString();
+		$skinResourcePatch = $this->getString();
+		$skinData = $this->getImage();
+		$animationCount = $this->getLInt();
+		$animations = [];
+		for($i = 0; $i < $animationCount; ++$i){
+			$animations[] = new SkinAnimation($this->getImage(), $this->getLInt(), $this->getLFloat());
+		}
+		$capeData = $this->getImage();
+		$geometryData = $this->getString();
+		$animationData = $this->getString();
+		$premium = $this->getBool();
+		$persona = $this->getBool();
+		$capeOnClassic = $this->getBool();
+		$capeId = $this->getString();
+		$fullSkinId = $this->getString();
+
+		return new Skin($skinId, $skinData->getData(), $capeData->getData(), $skinResourcePatch, $geometryData);
+	}
+
+	public function putSkin(Skin $skin){
+		$this->putString($skin->getSkinId());
+		$this->putString($skin->getGeometryName()); //resource patch
+		$this->putImage(SerializedImage::fromLegacy($skin->getSkinData()));
+		/** @var SkinAnimation[] $animations */
+		$animations = [];
+		$this->putLInt(count($animations));
+		foreach($animations as $animation){
+			$this->putImage($animation->getImage());
+			$this->putLInt($animation->getType());
+			$this->putLFloat($animation->getFrames());
+		}
+		$this->putImage(new SerializedImage(0, 0, $skin->getCapeData()));
+		$this->putString($skin->getGeometryData());
+		$this->putString(""); //animation data
+		$this->putBool(false); //isPremium
+		$this->putBool(false); //isPersona
+		$this->putBool(false); //isCapeOnClassic
+		$this->putString(""); //capeId
+		$this->putString(""); //fullskinId
+	}
+
+	public function getImage() : SerializedImage{
+		$width = $this->getLInt();
+		$height = $this->getLInt();
+		$data = $this->getString();
+		return new SerializedImage($height, $width, $data);
+	}
+
+	public function putImage(SerializedImage $image) : void{
```

Comment:
This should be renamed to `putSkinImage()` to avoid confusion.
It's also not clear to me that this needed to be public.